server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -68,6 +77,21 @@ public class RestoreSnapshotRequest extends MasterNodeRequest<RestoreSnapshotReq
     private boolean includeGlobalSettings = false;
     private String[] globalSettings = Strings.EMPTY_ARRAY;
 
+    private List<TableOrPartition> tablesToRestore = new ArrayList<>();
```

Comment:
```suggestion
    private List<TableOrPartition> tablesToRestore = List.of();
```

The instance is changed in the setter, and nobody should be adding elements to this.